spec.v2.yaml

```diff
@@ -0,0 +1,336 @@
+    # An empty list implies all active existing streams are supported.
+    # Requiring multiple streams at build time will result in multiple
+    # builds.  Requiring multiple streams at runtime implies the module
+    # is compatible with all of them.  If the same module streams are listed
+    # in both the build time and the runtime block, the build tools translate
+    # the runtime block so that it matches the stream the module was built
+    # against.  Multiple builds result in multiple output modulemd files.
+    # See below for an example.
+    # TODO: Provides, conflicts, obsoletes, recommends, etc.
+    #       Do we even need those?
+    # The example below illustrates how to build the same module in four
+    # different ways, with varying build time and runtime dependencies.
+    dependencies:
+        # Build on all available platforms except for f27, f28 and epel7
+        # After build, the runtime dependency will match the one used for
+        # the build.
+        - buildrequires:
+              platform: [-f27, -f28, -epel7]
+          requires:
+              platform: [-f27, -f28, -epel7]
+        # For platform:f27 perform two builds, one with buildtools:v2, another
+        # with buildtools:v2 in the buildroot.  Both will also utilize
+        # compatible:v3.  At runtime, buildtools isn't required and either
+        # compatible:v3 or compatible:v4 can be installed.
+        - buildrequires:
+              platform: [f27]
+              buildtools: [v1, v2]
+              compatible: [v3]
+          requires:
+              platform: [f27]
+              compatible: [v3, v4]
+        # For platform:f28 builds, require either runtime:a or runtime:b at
+        # runtime.  Only one build is performed.
+        - buildrequires:
+              platform: [f28]
+          requires:
+              platform: [f28]
+              runtime: [a, b]
+        # For platform:epel7, build against against all available extras
+        # streams and moreextras:foo and moreextras:bar.  The number of builds
+        # in this case will be 2 * <the number of extras streams available>.
+        # At runtime, both extras and moreextras will match whatever stream was
+        # used for build.
+        - buildrequires:
+              platform: [epel7]
+              extras: []
+              moreextras: [foo, bar]
+          requires:
+              platform: [epel7]
+              extras: []
+              moreextras: [foo, bar]
+        # Build dependencies of this module, optional
+        # Keys are module names, values are the stream names
+        # These modules define the buildroot for this module
+        buildrequires:
+            platform: and-its-stream-name
+            extra-build-env: and-its-stream-name-too
+        # Run-time dependencies of this module, optional
+        # Keys are module names, values are their stream names
+        requires:
+            platform: and-its-stream-name
```

Comment:
I suspect that this section was meant to be removed. As it exists right now, it is un-parseable because you are mixing a dictionary and a sequence at the same hierarchy level.